AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -3825,47 +3825,62 @@ function Debug-AzStorageAccountEntraKerbAuth {
         #
         # AAD Object check 
         #
-        if (!$filterIsPresent -or $Filter -match "CheckAADObject")
+        if (!$filterIsPresent -or $Filter -match "CheckEntraObject")
         {
             try {
                 $checksExecuted += 1;
-                Write-Verbose "CheckADObject - START"
+                Write-Verbose "CheckEntraObject - START"
                 $Context = Get-AzContext
                 $TenantId = $Context.Tenant
-                Connect-MgGraph -Scopes "Application.Read.All" -TenantId $TenantId
-                Get-MgApplication -Filter "identifierUris/any (uri:uri eq 'api://${TenantId}/CIFS/${StorageAccountName}.file.core.windows.net')" -ConsistencyLevel eventual
-                $SPNValue = Get-MgServicePrincipal -Filter "servicePrincipalNames/any (name:name eq 'api://$TenantId/CIFS/$StorageAccountName.file.core.windows.net')" -ConsistencyLevel eventual
-                if(-not $SPNValue.AccountEnabled)
+                Connect-MgGraph -Environment Global -Scopes "Application.Read.All" -TenantId $TenantId
+                $Application = Get-MgApplication -Filter "identifierUris/any (uri:uri eq 'api://${TenantId}/CIFS/${StorageAccountName}.file.core.windows.net')" -ConsistencyLevel eventual
+                if($null -eq $Application)
+                {
+                    $checks["CheckEntraObject"].Result = "Failed"
+                    $checks["CheckEntraObject"].Issue = "Could not find the application with SPN ' api://${TenantId}/CIFS/${StorageAccountName}.file.core.windows.net'."
+                    Write-Error "CheckEntraObject - FAILED"
+                    Write-Error "Could not find the application with SPN 'api://${TenantId}/CIFS/${StorageAccountName}.file.core.windows.net' "
+                }
+                $ServicePrincipal = Get-MgServicePrincipal -Filter "servicePrincipalNames/any (name:name eq 'api://$TenantId/CIFS/$StorageAccountName.file.core.windows.net')" -ConsistencyLevel eventual
+                if($null -eq $ServicePrincipal)
+                {
+                    $checks["CheckEntraObject"].Result = "Failed"
+                    $checks["CheckEntraObject"].Issue = "Service Principal is missing SPN ' CIFS/${StorageAccountName}.file.core.windows.net'."
+                    Write-Error "CheckEntraObject - FAILED"
+                    Write-Error "SPN Value is not set correctly, It should be 'CIFS/Storageaccountname.file.core.windows.net'"
+                }
+                }
+                if(-not $ServicePrincipal.AccountEnabled)
                 {
-                    $checks["CheckADObject"].Result = "Failed"
-                    $checks["CheckADObject"].Issue = "SPN Value is not Set correctly."
-                    Write-Error "CheckADObject - FAILED"
+                    $checks["CheckEntraObject"].Result = "Failed"
+                    $checks["CheckEntraObject"].Issue = "SPN Value is not Set correctly."

```

Comment:
This error message should say: "Expected AccountEnabled to be set to tue"

Reply:
fixed